src/tasks/installers/common-tasks.ts

```diff
@@ -155,6 +155,11 @@ export function createEclipseCheCluster(flags: any, kube: KubeHelper): Listr.Lis
   }
 }
 
+function shouldNotDeployKeycloak(flags: any, cr: any): boolean {
+  return cr.spec.auth.externalIdentityProvider || cr.spec.auth.nativeUserMode ||
+    (isOpenshiftPlatformFamily(flags.platform) && cr.spec.devWorkspace.enable)
+}
+
```

Comment:
Yes, that's a common issue when operator modifies checluster and chectl is not aware of that.
But in this case it leads to another problem. If chectl is used to deploy a previous version of Eclipse Che (--version flag) then keycloak deployment won't be handled. 

So, I suggest to modify [1] to pass some function to check if `nativeUserMode` is on and skip a task in this case.

[1] https://github.com/che-incubator/chectl/blob/main/src/tasks/kube.ts#L31
[2] https://github.com/che-incubator/chectl/blob/main/src/tasks/che.ts#L257
[3] https://github.com/che-incubator/chectl/blob/main/src/tasks/che.ts#L96

Reply:
Done